### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by jburel**
Move the setup to a separate file and clean up the code
it is no longer required to checkout the sha since the workflow is triggered when a branch is merged to master

### CODE REVIEW

File: .github/workflows/setup.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+MS_SUITE="omero-ms-zarr-suite"
+BF_SUITE="ome-zarr-bf2raw-suite"
+ZARR_SUITE="ome-zarr-py-suite"
+
+REPO=$1
+# Checkout the sha from the specified repository.
```
Comment (joshmoore) [inline .github/workflows/setup.sh:6]: This is done by configure_test.py? Or can this comment be dropped?

**Review by joshmoore — COMMENTED** [review]
One minor question. There are probably simplifications that we can add, but I'd say let's handle those as we add more repos here. :+1:

File: .github/workflows/setup.sh (hunk 1 above)
Comment (jburel) [reply .github/workflows/setup.sh:6]: Actually this is no longer needed since we fetch head.
Thanks

**Comment by joshmoore** [comment]
Cheers, thanks, @jburel

**Comment by sbesson** [comment]
This commit seemed to have introduced a regression in the dispatch workflow where is it executed causing builds to fail and false positive issues to be opened - see https://github.com/ome/ome_zarr_test_suite/actions/runs/1546553644 and  https://github.com/ome/ome_zarr_test_suite/actions/runs/1546473294 for recent examples

```
Run # Set the parameters to be used in the response
  # Set the parameters to be used in the response
  repo=ome/ome-zarr-py
  owner=ome
  sha=
  repo_name=${repo#"$owner"}
  repo_name=${repo_name#"/"}
  echo "client_repo_name="$repo_name >> $GITHUB_ENV
  echo "client_sha="$sha >> $GITHUB_ENV
  ./setup.sh $repo
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    ms_suite: omero-ms-zarr-suite
    bf_suite: ome-zarr-bf2raw-suite
    zarr_suite: ome-zarr-py-suite
    CONDA_PKGS_DIR: /home/runner/conda_pkgs_dir
/home/runner/work/_temp/4f9c9013-ffbf-4d13-bef3-fe281cae0517.sh: line 9: ./setup.sh: No such file or directory
```

I assume `working-directory` needs to be set accordingly?

**Comment by jburel** [comment]
Fixed

**Pull request merged**